include/onnx-mlir/Runtime/OMTensor.h

```diff
@@ -115,11 +115,15 @@ OM_EXTERNAL_VISIBILITY OMTensor *omTensorCreate(
  * @param shape list of integers indicating the tensor shape.
  * @param rank tensor rank.
  * @param dtype tensor element data type.
- * @param owning whether OMTensor owns the data, if set to true, OMTensor will
- * release the data_ptr upon destruction.
+ * @param owning whether OMTensor owns the data, if set to
+ * OMTENSOR_OWNING_DATA_PTR, OMTensor will release the data_ptr upon
+ * destruction. If set to OMTENSOR_OWNING_DATA_PTR_AND_STRING, OMTensor will
+ * free both the data_ptr and its corresponding strings upon destruction.
  * @return pointer to OMTensor created, NULL if creation failed.
  *
  */
+#define OMTENSOR_OWNING_DATA_PTR 1
+#define OMTENSOR_OWNING_DATA_PTR_AND_STRING 2
```

Comment:
Your change can support pointers to data other than string, right?

Reply:
Current implementation stores char * pointer in the Tensor of string. If a Tensor of string and the strings are created inside the compiler, we will have to allocate the space for data_ptr and for the strings.  That will be a new case for the ownership. If the Tensor of strings are just propagate pointers from one tensor to another, the ownership will be the same as the float or integer data type.

Reply:
@gongsu832 Thanks for the comments. I changed the code according to your comments. I updated the code to use the OMTensor->_allocated buffer for string data themselves. And there are no modifications on src/Runtime/OMTensor.inc from the original code now.

The string data by allocated by Python applications are copied into memory area in the OMTensor->_allocated buffer, which follows the area for array of string pointers (in the PyExecutionSessionBase::pyRun function in PyExecutionSessionBase.cpp). The _allocated buffer is freed in the mTensorDestroy like other cases.

I hope that this update can answer your concerns related to the OMTensor->_owning flag.
Please let me know your additional comments and suggestions.

(Let me explain my thought about the multi-dimensional array in Python by another comments.)